ui/index.d.ts

```diff
@@ -32,6 +32,32 @@ export declare class AzureTreeDataProvider implements TreeDataProvider<IAzureNod
     public findNode(id: string): Promise<IAzureNode | undefined>;
     public dispose(): void;
 }
+/**
+ * The TestAzureTreeDataProvider is an implementation of the IAzureTreeDataProvider that automatically signs into
+ * an Azure account via service principals with credentials stored on Travis.  Because signing in is an interactive
+ * experience, this circumvents that to do automated testing with Azure resources.
+ */
+export declare class TestAzureTreeDataProvider implements IAzureTreeDataProvider<IAzureNode>, Disposable {
```

Comment:
It would be much better to share the bulk of this code.  I'm thinking having a test version of the azure account extension interface might be the best way to do it.  I just fixed some bugs in this code, for instance, we shouldn't need to make such changes in two places.

Reply:
I think that this makes way more sense.  I'll try to implement it in that way and resubmit a PR.